universal_transfer_operator/transfers/datasets/base.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+from urllib.parse import urlparse
+
+from airflow.utils.log.logging_mixin import LoggingMixin
+from attr import define, field
+
+try:
+    # Airflow >= 2.4
+    from airflow.datasets import Dataset
+
+    DATASET_SUPPORT = True
+except ImportError:
+    # Airflow < 2.4
+    Dataset = object
+    DATASET_SUPPORT = False
+
+
+@define
+class UniversalDataset(LoggingMixin, Dataset):
+    """
+    Repersents all file dataset, and abstract away the details related to location and file types.
+    Intended to be used within library.
+
+    :param path: Path to a file in the filesystem/Object stores
+    :param conn_id: Airflow connection ID
+    """
+
+    path: str
+    conn_id: str
```

Comment:
shouldn't be required

Reply:
Removed it: https://github.com/astronomer/astro-sdk/pull/1492/commits/b59f087700a88b7f24989d0619c89bd06771c8e5